### PR TITLE
Fix `selector-pseudo-element-no-unknown` false positives for `view-transition` pseudo-elements

### PR DESCRIPTION
**Pull request by danielroe**
> Which issue, if any, is this issue related to?

Resolves https://github.com/stylelint/stylelint/issues/7069

> Is there anything in the PR that needs further explanation?

No, it's self-explanatory. I've added some details in the linked issue.

### CODE REVIEW

File: .changeset/lazy-geese-begin.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"stylelint": patch
+---
+
+Add support for view-transition pseudo-elements
```
Comment (jeddy3) [inline .changeset/lazy-geese-begin.md:5]:
```suggestion
Fixed: `selector-pseudo-element-no-unknown` false positives for `view-transition` pseudo-elements
```

To follow [our conventions](https://stylelint.io/maintainer-guide/pull-requests).

**Review by jeddy3 — APPROVED** [review]
LGTM, thank you!

**Comment by danielroe** [comment]
🤝

**Review by ybiquitous — APPROVED** [review]
@danielroe Thanks for your first-time contribution! LGTM 👍🏼 

I've suggested a minor point, but we can merge this as it is.

File: lib/reference/selectors.js [hunk 2]
```diff
@@ -177,6 +177,11 @@ const pseudoElements = uniteSets(
 		'slotted',
 		'spelling-error',
 		'target-text',
+		'view-transition',
+		'view-transition-group',
+		'view-transition-image-pair',
+		'view-transition-old',
+		'view-transition-new',
```
[the remark this answers is not in the document]
Comment (danielroe) [reply lib/reference/selectors.js:184]: Quite right.

File: lib/reference/selectors.js [hunk 3]
```diff
@@ -177,6 +177,11 @@ const pseudoElements = uniteSets(
 		'slotted',
 		'spelling-error',
 		'target-text',
+		'view-transition',
+		'view-transition-group',
+		'view-transition-image-pair',
+		'view-transition-old',
```
Comment (danielroe) [inline lib/reference/selectors.js:183]:
```suggestion
		'view-transition-new',
```

**Review by ybiquitous — APPROVED** [review]
Thank you! 👍🏼

**Pull request merged**